Sources/Foundation/FileManager.swift

```diff
@@ -579,13 +579,13 @@ open class FileManager : NSObject {
 #elseif os(WASI)
         let type = FileAttributeType(statMode: mode_t(s.st_mode))
 #else
-        if let pwd = getpwuid(s.st_uid), pwd.pointee.pw_name != nil {
-            let name = String(cString: pwd.pointee.pw_name)
+        if let pwd = getpwuid(s.st_uid), let pwd_name = pwd.pointee.pw_name {
+            let name = String(cString: pwd_name)
```

Comment:
```suggestion
        if let pwd = getpwuid(s.st_uid), let pw_name = pwd.pointee.pw_name {
            let name = String(cString: pw_name)
```

Reply:
Change is fine by me, but as mentioned in #4889, I will get this working with the `swift-foundation` re-core by next week and re-submit for trunk then.

Reply:
https://github.com/apple/swift-corelibs-foundation/pull/5011 is the PR to track merging main to release/6.0.